src/Product/WCProductAdapter.php

```diff
@@ -1070,4 +1077,24 @@ protected function format_attribute( $value, $attribute_id ) {
 
 		return $value;
 	}
+
+	/**
+	 * Gets a custom attribute from a product
+	 *
+	 * @param string $attribute_name - The attribute name to get.
+	 * @return string|null The attribute value or null if no value is found
+	 */
+	protected function get_custom_attribute( $attribute_name ) {
+		$product = $this->get_wc_product();
+
+		$attribute_values = $product->get_attribute( $attribute_name );
+
+		if ( ! $attribute_values ) {
+			$attribute_values = $product->get_meta( $attribute_name );
+		}
+
+		$attribute_values = explode( ', ', $attribute_values );
```

Comment:
So these custom attributes are never taxonomies. These are the attributes saved on the fly in the product or any other custom field (like the ones you can set by checking Screen -> Custom fields) and should be loaded by the user using a filter.  In the last ones (custom fields) they are not returned by `get_attribute` that's why `get_meta` was in place.

<img width="479" alt="Screenshot 2023-05-04 at 17 29 54" src="https://user-images.githubusercontent.com/5908855/236228669-eb977e38-905c-4984-b056-dd2594401670.png">

<img width="1053" alt="Screenshot 2023-05-04 at 17 54 42" src="https://user-images.githubusercontent.com/5908855/236229130-72f4c159-5f3c-4470-9e71-738ce7cd846b.png">

<img width="1258" alt="Screenshot 2023-05-04 at 17 54 20" src="https://user-images.githubusercontent.com/5908855/236228769-771f4651-2432-45b1-8d55-1b4c4a1fc042.png">

Example: 
```
add_filter(
		'woocommerce_gla_attribute_mapping_sources_custom_attributes',
		function( $values ) {
			return array_merge( $values, ['isbn', 'logo']);
		}
	);
```

The recommendation in the placeholder is to separate those by "|" which is transformed by WC into " | ". So we can try to do so in our code and also check for the meta being a scalar in order to prevent errors. You can see the adapted tests and the refactor in the last commits.